R/bundle.R

```diff
@@ -485,7 +502,11 @@ addPackratSnapshot <- function(bundleDir, implicit_dependencies = c()) {
 
   # generate the packrat snapshot
   tryCatch({
-    performPackratSnapshot(bundleDir)
+    if (!file.exists(snapshotLockFile(bundleDir))) {
+      performPackratSnapshot(bundleDir)
+    } else {
+      message("Re-using Packrat Lock File")
```

Comment:
Do we need to check that the lockfile is up to date before using it? i.e. what if my packrat status isn't clean (I'm using packages I haven't yet snapshotted) and I try to publish?